integration-tests/modules/__tests__/product/store/index.spec.ts

```diff
@@ -0,0 +1,332 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  CreateProductDTO,
+  IPricingModuleService,
+  IProductModuleService,
+  ISalesChannelModuleService,
+  ProductDTO,
+  ProductVariantDTO,
+} from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  Modules,
+  ProductStatus,
+} from "@medusajs/utils"
+import { medusaIntegrationTestRunner } from "medusa-test-utils"
+import { createAdminUser } from "../../../../helpers/create-admin-user"
+import { createDefaultRuleTypes } from "../../../helpers/create-default-rule-types"
+import { createVariantPriceSet } from "../../../helpers/create-variant-price-set"
+
+jest.setTimeout(50000)
+
+const env = { MEDUSA_FF_MEDUSA_V2: true }
+const adminHeaders = { headers: { "x-medusa-access-token": "test_token" } }
+
+async function createProductsWithVariants(
+  productModule: IProductModuleService,
+  productsData: CreateProductDTO
+): Promise<[ProductDTO, ProductVariantDTO[]]> {
+  const { variants: variantsData, ...productData } = productsData
+
+  const [product] = await productModule.create([productData])
```

Comment:
Creating the product like this is problematic, as there are things that run in workflows that won't get executed. We should try and call the `admin` HTTP endpoint instead. There you will also just pass the `variants` as part of the product, no need for 2 separate calls